public/app/core/history/RichHistoryRemoteStorage.ts

```diff
@@ -132,11 +134,11 @@ function buildQueryParams(filters: RichHistorySearchFilters): string {
     params = params + `&sort=${filters.sortOrder === SortOrder.Ascending ? 'time-asc' : 'time-desc'}`;
   }
   if (!filters.starred) {
-    const relativeFrom = filters.from === 0 ? 'now' : `now-${filters.from}d`;
-    const relativeTo = filters.to === 0 ? 'now' : `now-${filters.to}d`;
     // TODO: Unify: remote storage from/to params are swapped comparing to frontend and local storage filters
-    params = params + `&to=${relativeFrom}`;
-    params = params + `&from=${relativeTo}`;
+    const from = createRetentionPeriodBoundary(filters.to ?? 14, { isLastTs: true, tz: contextSrv.user?.timezone });
```

Comment:
There is also a timezone in the Explore redux state but we still also keep it updated in the contextSrv so I thought that should be treated as the main source of truth here (also it's easier to access here)